forte/data/selector.py

```diff
@@ -66,12 +68,32 @@ def select(self, pack: MultiPack) -> Iterator[DataPack]:
 class NameMatchSelector(SinglePackSelector):
     r"""Select a :class:`DataPack` from a :class:`MultiPack` with specified
     name.
+
+    This implementation takes special care for backward compatability:
+    Deprecated:
+        selector = NameMatchSelector(select_name="foo")
+        selector = NameMatchSelector("foo")
+    Now:
+        selector = NameMatchSelector(
+            configs={
+                "select_name": "foo"
+            }
+        )
     """
 
-    def __init__(self, select_name: str):
-        super().__init__()
-        assert select_name is not None
-        self.select_name: str = select_name
+    def __init__(self, *args, **kwargs):
+        assert (len(args) == 0) ^ (len(kwargs) == 0)
```

Comment:
hmm, I think raising valueerror is better than assert, and we can provide clear error message on this.

Reply:
agree